lib/resty/upstream/healthcheck.lua

```diff
@@ -92,6 +93,92 @@ local function set_peer_down_globally(ctx, is_backup, id, value)
     end
 end
 
+local function get_peers_to_check(peers)
+    local peers_to_check = {}
+    for _, peer in ipairs(peers) do
+        if not checked_peers_upstream[peer.name] then
+            table.insert(peers_to_check, peer)
+        end
+    end
+    return peers_to_check
+end
+
+local function get_checked_peers(peers)
+    local checked_peers = {}
+    for _, peer in ipairs(peers) do
+        if checked_peers_upstream[peer.name] then
+            table.insert(checked_peers, peer)
+        end
+    end
+    return checked_peers
+end
+
+local function get_all_upstream_peers(u)
+    local primary_peers, err = upstream.get_primary_peers(u)
+    if not primary_peers then
+        errlog("failed to get primary peers: ", err)
+        return
+    end
+
+    local backup_peers, err = upstream.get_backup_peers(u)
+    if not backup_peers then
+        errlog("failed to get backup peers: ", err)
+        return
+    end
+
+    return primary_peers, backup_peers
+end
+
+local function get_unique_upstream_peers(u)
+    primary_peers, backup_peers = get_all_upstream_peers(u)
+
+    local ppeers_to_check = get_peers_to_check(primary_peers)
+    local bpeers_to_check = get_peers_to_check(backup_peers)
+
+    return ppeers_to_check, bpeers_to_check
+end
+
+local function already_checked_peers(u)
+    primary_peers, backup_peers = get_all_upstream_peers(u)
+
+    local ppeers_checked = get_checked_peers(primary_peers)
+    local bpeers_checked = get_checked_peers(backup_peers)
+
+    return ppeers_checked, bpeers_checked
+end
+
+local function set_status(peers, is_backup, ctx)
+    local dict = ctx.dict
+    local u = ctx.upstream
+
+    for i = 1, #peers do
+        local peer = peers[i]
+        local already_set = dict:get(gen_peer_key("ok:", u, is_backup, peer.id))
+        if already_set then
+            return
+        end
+
+        local checked_peer_upstream = checked_peers_upstream[peer.name]
+
+        local oks, err = dict:get(gen_peer_key("ok:", checked_peer_upstream, is_backup, peer.id))
```

Comment:
What happens if `err` is set here? If it matters, do something with it. If it doesn't, then `local oks, _ = ...`